File: library/src/main/java/de/westnordost/osmfeatures/Feature.java
package de.westnordost.osmfeatures;

import java.util.List;
import java.util.Locale;
import java.util.Map;

/** Subset of a feature as defined in the iD editor
 *  https://github.com/ideditor/schema-builder#preset-schema
 *  with only the fields helpful for the dictionary */
public interface Feature {
    String getId();
    Map<String,String> getTags();
    List<GeometryType> getGeometry();
    String getName();
    String getIcon();
    String getImageURL();
    List<String> getTerms();
    List<String> getIncludeCountryCodes();
    List<String> getExcludeCountryCodes();
    boolean isSearchable();
    double getMatchScore();
    Map<String,String> getAddTags();
    Map<String,String> getRemoveTags();

    String getCanonicalName();
    List<String> getCanonicalTerms();

    Locale getLocale();
}

// currently not included:
// - fields
// - moreFields
// - replacement
// - reference

File: library/src/commonMain/kotlin/de.westnordost.osmfeatures/Feature.kt
package de.westnordost.osmfeatures

/** Subset of a feature as defined in the iD editor
 *  https://github.com/ideditor/schema-builder#preset-schema
 *  with only the fields helpful for the dictionary */
interface Feature {
    val id: String
    val tags: Map<String, String>
    val geometry: List<GeometryType>
    val name: String
    val icon: String?
    val imageURL: String?

    /** name + aliases  */
    val names: List<String>
    val terms: List<String>
    val includeCountryCodes: List<String>
    val excludeCountryCodes: List<String>
    val isSearchable: Boolean
    val matchScore: Float
    val addTags: Map<String, String>
    val removeTags: Map<String, String>
    val canonicalNames: List<String>
    val canonicalTerms: List<String>?
    val isSuggestion: Boolean
    val locale: Locale?
}
